### ISSUE TITLE
Allow tileUrlFunction to be called on small resolution changes (even when gridZoom {z} remains constant)

### ISSUE THREAD

**Issue by Biosphakus** [openlayers/openlayers#10432]
I'm trying to load tiles from different subdomains of a custom XYZ source depending on the current resolution (viewZoom). Different resolutions of the view might be associated with different subdomains but with identical gridZoom. In this case, the ``tileUrlFunction`` cannot be called.

```
viewZoom zoom   subdomain
0        1.0    subdomain1
1        1.2    subdomain1
2        1.4    subdomain2
3        1.6    subdomain3

gridZoom = Math.round(zoom);
``` 

Several approaches exist to enable loading from different sources (see [here](https://gis.stackexchange.com/questions/344604/openlayers-smoothly-change-tile-source-on-zoom/345346#345346) for a more detailed description) but they seem to be lacking some of the advantages of the ``tileUrlFunction`` (before zoom: preload adjacent zoom levels; on zoom: resize old tiles, then load new tiles, then remove old tiles). 

**Use case:**
Semantic zooming, map with several levels of detail which might be shown at intermediate zoom steps without changing the scale enough to alter gridZoom (e. g. administrative borders at a continental, national and regional scale; my use case is a biological map where I try to show different steps of a taxonomic hierarchy).

**Comment by mike-000** [comment]
It could be done using a custom tile grid which specifies the resolutions and tile sizes you would like and whatever is served fits into those https://codesandbox.io/s/simple-bg7d4

**Comment by Biosphakus** [comment]
Thanks a lot for your answer! In principle, this is exactly what I want and it seems to be just as flexible as the standard tile grid. However, in some cases it doesn't quite seem to work. Sometimes, zoom-in with subsequent panning doesn't replace old tiles. The problem remains even after panning.
![tile_error](https://user-images.githubusercontent.com/58983561/71315076-2d5b4e00-2455-11ea-83ad-17140c0e8f4d.png)

Something else I couldn't figure out yet: When I set a `transition` in the layer, new tiles will load very smoothly but going back to the old tiles, which were already loaded, replaces them instantly. Is there any possibility to make them fade in smoothly for both cases?

**Comment by mike-000** [comment]
Tiles from services such such as OSM and MapTiler sometimes fail to update if the server is busy (and browser caching means the problem can recur even after closing and reopening the map).  When zoom levels have similar styles you might not notice, but with different styles it will be more noticeable - you could try setting `useInterimTilesOnError: false` as a layer option to prevent display of the old tiles. 

With a vector layer you can change styles (or set empty style to hide features) using the style function's resolution parameter, but if you need to clear and reload the data you would need to listen for `change:resolution` events on the view or use multiple layers (possibly with `minResolution` and `maxResolution` options).

**Comment by Biosphakus** [comment]
Thank you! [Here](https://gis.stackexchange.com/questions/199333/how-to-force-a-ol-source-vector-to-reload-data-from-server), I've found a very nice approach for the vector layer too. 

It works well for both with only minor quirks that are related to the transition between zoom levels. Keeping tiles from lower zoom levels until the new ones are loaded does seem to be preferable on zoom-in. However, on zoom-out and panning I'd prefer not to show any tiles before the new ones are loaded (like [here](https://gis.stackexchange.com/questions/261269/hide-old-tiles-while-loading-new-ones-when-zooming-in-openlayers), but only on zoom-out and panning). 

In the vector layer, I'm using vector labels with a size corresponding to the zoom level to show some text. Its style is updated on zoom changes (which is great) but only after the zoom is completed. I would prefer to have it updated on some intermediate steps as well. Can this behaviour be modified in any way? (However, that's minor and I'm already quite happy with the solution I've got.)
